### PR TITLE
FEAT: Upgrade RAY cluster with latest structure

### PR DESCRIPTION
**Pull request by umeshkumhar**
- Enable GCS csi flag in infrastrcuture 
- move ray into new structure

### CODE REVIEW

**Review by imreddy13 — APPROVED** [review]

File: applications/ray/README.md
Comment (imreddy13) [inline applications/ray/README.md:1]: Please fix line 6 to reflect how to create a cluster (from /infrastructure and not /gke-platform)

File: applications/jupyter/workloads.tfvars [hunk 1]
```diff
@@ -17,30 +17,31 @@
...
 ## JupyterHub variables
-namespace = "ml"
-gcs_bucket = "gcs-bucket"
-k8s_service_account = "default"
-gcp_service_account ="jupyter-service-account1"
+namespace                     = "ml"
+gcs_bucket                    = "gcs-bucket"
+create_k8s_service_account    = false
+k8s_service_account           = "default"
+gcp_service_account           = "jupyter-service-account1"
```
Comment (imreddy13) [inline applications/jupyter/workloads.tfvars:33]: remove the "1"

File: applications/jupyter/workloads.tfvars [hunk 2]
```diff
@@ -17,30 +17,31 @@
...
 
 ## JupyterHub variables
-namespace = "ml"
-gcs_bucket = "gcs-bucket"
-k8s_service_account = "default"
-gcp_service_account ="jupyter-service-account1"
+namespace                     = "ml"
+gcs_bucket                    = "gcs-bucket"
+create_k8s_service_account    = false
+k8s_service_account           = "default"
```
Comment (imreddy13) [inline applications/jupyter/workloads.tfvars:32]: All these will need to be fixed up again once my SA PR goes in fyi

File: applications/ray/example_notebooks/jax-tpu.ipynb
Comment (imreddy13) [inline applications/ray/example_notebooks/jax-tpu.ipynb:1]: is it easy to make example_notebooks a top level folder? like ai-on-gke/example_notebooks ?

File: infrastructure/platform.tfvars [hunk 3]
```diff
@@ -19,13 +19,13 @@ project_id = "ai-on-gke-jss-sandbox"
...
 ## network values
-create_network  = true
+create_network  = false
 network_name    = "ml-network"
 subnetwork_name = "ml-subnet"
 
 ## required only in case new network provisioning
 subnetwork_cidr           = "10.100.0.0/16"
-subnetwork_region         = "us-central1"
+subnetwork_region         = "us-central"
```
Comment (imreddy13) [inline infrastructure/platform.tfvars:28]: is this correct?

File: modules/kuberay-serviceaccounts/service_accounts.tf [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+resource "google_project_iam_binding" "monitoring-viewer-sa" {
+  project = var.project_id
+  role    = "roles/iam.serviceAccountTokenCreator"
+
+  members = [
+    "serviceAccount:${google_service_account.sa_prom.account_id}@${var.project_id}.iam.gserviceaccount.com",
+  ]
+}
+
+resource "kubernetes_service_account" "ksa-prom" {
```
Comment (imreddy13) [inline modules/kuberay-serviceaccounts/service_accounts.tf:48]: what does this do?

File: modules/jupyter_service_accounts/service_accounts.tf [hunk 5]
```diff
@@ -23,28 +23,40 @@ resource "google_service_account_iam_binding" "sa-binding" {
...
-  for_each = toset(var.sa_iam_roles)
+  for_each = toset(var.gcp_sa_iam_roles)
   role = each.key
 
   members = [
     "serviceAccount:${google_service_account.sa.account_id}@${var.project_id}.iam.gserviceaccount.com",
   ]
 }
 
+resource "kubernetes_service_account" "ksa" {
```
Comment (imreddy13) [inline modules/jupyter_service_accounts/service_accounts.tf:40]: I got an error that kubernetes_service_account usage with the service account token automount is deprecated after 1.24 k8s - did you also see this? May need to use kubernetes_manifest

File: modules/jupyter/main.tf [hunk 6]
```diff
@@ -61,13 +61,15 @@ module "iap_auth" {
 }
 
 module "workload_identity_service_account" {
-  source = "../../modules/service_accounts"
-
-  project_id      = var.project_id
-  namespace       = var.namespace
-  service_account = var.gcp_service_account
-  sa_iam_roles    = split(",", var.gcp_service_account_iam_roles)
```
Comment (imreddy13) [inline modules/jupyter/main.tf:69]: can we not use a list[] for this?

File: applications/jupyter/workloads.tfvars [hunk 7]
```diff
@@ -17,30 +17,31 @@
...
+gcp_service_account           = "jupyter-service-account1"
 gcp_service_account_iam_roles = "roles/storage.admin,roles/artifactregistry.reader"
```
Comment (imreddy13) [inline applications/jupyter/workloads.tfvars:34]: can this be a list?

File: infrastructure/platform.tfvars (hunk 3 above)
Comment (umeshkumhar) [reply infrastructure/platform.tfvars]: fixed

**Review by imreddy13 — APPROVED** [review]

**Pull request merged**